meson.build

```diff
@@ -5,13 +5,21 @@ project(
 	'c',
 	version: '0.0.0',
 	meson_version: '>= 0.63',
+	default_options: [
+		'c_std=gnu99',
+	]
 )
 
 c = meson.get_compiler('c')
-if c.get_id() == 'msvc'
+if target_machine.system() == 'windows'
```

Comment:
https://mesonbuild.com/Cross-compilation.html

host_machine is the cross compile target.

target_machine is rarely ever needed, but when it is explicitly specified in a cross file it means that the meson project is cross compiling a program for the host machine, and that program is, itself, a compiler, and that compiler program is not just any compiler but in fact a cross compiler.

Example: build GCC on a 64-bit Linux buildbot (build_machine is linux x86_64), using mingw to cross compile a Windows program (host_machine is windows x86_64), but the end GCC is going to be installed on a Windows machine and be used by Windows users to cross compile raspberry pi programs (target_machine is linux aarch64).

By default, target_machine evaluates the same as host_machine since a compiler that supports [Canadian Cross](https://en.wikipedia.org/wiki/Cross_compiler#Canadian_Cross) might also be used to cross compile a pure-native (two machines) compiler, in which case target == host.

AFAICT you're not using the target machine here, just the host machine.

Reply:
Again, the distinction I'm making here is that for the cross-compilation case you want host_machine.

For the "Canadian Cross: I am cross-compiling a cross-compiler" case, you want target_machine for whenever you need to refer to the third machine


> Using MinGW to compile for Windows on Linux, for example, will still emit the warnings we want disabled here.

- Using mingw on Linux: this is build_machine
- to compile for Windows: this is the host_machine

Which machine is the target_machine here? Once you build your libcuefile.dll and copy it for deploying onto Windows machines, a target_machine would imply that libcuefile, running on Windows, emits output files that are intended for use on not-Windows (perhaps they are supposed to run on macOS, or Android, or a raspberry pi).

target_machine defaults to host_machine, so this technically works either way. It's just *confusing* to see target_machine being used outside of rare situations.

Reply:
oh, we see what you're trying to say.. hmm.. very confusingly named as we talk about "the target machine of the binary" a lot in embedded systems vs a host machine. Thanks for the review feedback and we'll work with @amyspark in updating several projects to use host_machine then w/ a note that we're not talking about the machine compilation is hosted on but the machine the output targets.